### PR TITLE
Adapt code to spatstat v2

### LINKED ISSUE #137: CRAN errors with spatstat 2.0-1
**Describe the bug**
We got an e-mail from CRAN with our first errors!

> Dear maintainer,
> 
> Please see the problems shown on
> <https://cran.r-project.org/web/checks/check_results_sfnetworks.html>.
> 
> The errors come from spatstat 2.0-1
> 
> Please correct before 2021-03-28 to safely retain your package on CRAN.
> 
> The CRAN Team

It seems like we need to link to the new `spatstat` sub-packages specifically for the methods and examples implemented (?) We [noticed that `sf` might be dealing with the same issues](https://github.com/r-spatial/sf/commit/8fb16e899222241302bd27f7c6811457dbfc735d#diff-9cc358405149db607ff830a16f0b4b21f7366e3c99ec00d52800acebe21b231c) which might give some hints on how to solve this. @agila5 could you look into this?

**Reproducible example**
Check CRAN results.

**Expected behavior**
Success!

### PR DESCRIPTION
**Pull request by agila5**
Fix #137. 

Finally, it seems that it's working. I will complete the PR as soon as `sf` 0.9.8 is on CRAN (the 0.9.7 version is [already failing](https://cran.r-project.org/web/checks/check_results_sf.html)). The only downside is that (I think) we have to add a dependency on the 0.9.8 version of `sf` or, at least, test the `sf` version in the spatstat-related code. Do you think it's a problem?

### CODE REVIEW

**Comment by agila5** [comment]
I think it's ready to be reviewed/merged. 

IMO the only problem is that it introduces a "strong" dependency on `sf >= 0.9.8` (and also an ugly startup message related to `spatstat.geom`). Hence, I have the following proposal. I am probably the only one that takes advantage of the integration between `sfnetworks` and `spatstat`. So, maybe, we could purge all `spatstat` code from `sfnetworks`, and then I will create a new (small but with a lot of dependencies) package that extends `sfnetworks`. Opinions?

**Comment by luukvdmeer** [comment]
> I am probably the only one that takes advantage of the integration between sfnetworks and spatstat

Not sure about that! I think it is great to have integration with other packages. I was planning to implement similar integration with e.g. `spdep`. 

What is the reason we need the sf dependency? I don't see sf functions being called in your code (but might have looked over them). If we need it, I would suggest to check for the sf version in the `check_spatstat` function, rather than setting the depedency for the whole package.

Regarding the startup message, you refer to this one?

```
Registered S3 method overwritten by 'spatstat':
  method     from
  print.boxx cli 
```

I think we can call `s3_register("spatstat.linnet::as.linnet", "sfnetwork")` inside `suppressPackageStartupMessages()` to avoid that, but not sure if that is good practice.

File: R/spatstat.R [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    spst_ver <- try(packageVersion("spatstat"), silent = TRUE)
+    if (!inherits(spst_ver, "try-error") && spst_ver < 2.0-0) {
+      stop(paste0(
+        "You have an old version of spatstat installed which is incompatible with ",
+        pkg,
+        ". Please update spatstat (or uninstall it)."
+      ))
+    }
+  }
+}
```
Comment (luukvdmeer) [inline R/spatstat.R:19]: Can we include this function in the `linnet.R` file (and maybe rename that file to `spatstat.R` instead?)

File: R/zzz.R [hunk 2]
```diff
@@ -7,7 +7,13 @@
   if (has_ggplot2_3.0)
     s3_register("ggplot2::autoplot", "sfnetwork")
 
-  s3_register("spatstat::as.linnet", "sfnetwork")
+  has_spatstat_2.0 =
+    requireNamespace("spatstat", quietly = TRUE) &&
+    utils::packageVersion("spatstat") >= "2.0.0"
+
+  if (has_spatstat_2.0) {
+    s3_register("spatstat.linnet::as.linnet", "sfnetwork")
```
Comment (luukvdmeer) [inline R/zzz.R:15]: In think here we could use:

```
  if (has_spatstat_2.0) {
    suppressPackageStartupMessages(
      s3_register("spatstat.linnet::as.linnet", "sfnetwork")
    )
  }
```

such that now spatstat related startup messages are printed

File: R/spatstat.R (hunk 1 above)
Comment (agila5) [reply R/spatstat.R:19]: Fixed with the last commits!

File: R/zzz.R (hunk 2 above)
Comment (agila5) [reply R/zzz.R]: Fixed with the last commits!

**Comment by agila5** [comment]
> What is the reason we need the sf dependency? I don't see sf functions being called in your code (but might have looked over them).

The method for converting `sf` objects (with `POINT` geometry) into [`ppp`](https://rdrr.io/cran/spatstat.geom/man/ppp.object.html) objects is defined [here](https://github.com/r-spatial/sf/blob/80d6a1999929cd133b1fef77eb5a72a88a56c246/R/spatstat.R#L127-L140) and used here

https://github.com/luukvdmeer/sfnetworks/blob/236318a132f1d52f8bb93a00c0d6a72c155324e9/R/spatstat.R#L54-L55

The method for converting [`psp`](https://rdrr.io/cran/spatstat.geom/man/psp.object.html) (and [`linnet`](https://rdrr.io/cran/spatstat.linnet/man/linnet.html)) objects into `sf` objects is defined [here](https://github.com/r-spatial/sf/blob/80d6a1999929cd133b1fef77eb5a72a88a56c246/R/spatstat.R#L56-L72) and used here

https://github.com/luukvdmeer/sfnetworks/blob/236318a132f1d52f8bb93a00c0d6a72c155324e9/R/sfnetwork.R#L308-L311

The problem is that the code in the `sf` package version 0.9.7 fails because it calls `spatstat` functions that were transferred to a different package. 

> If we need it, I would suggest to check for the sf version in the check_spatstat function, rather than setting the depedency for the whole package.

Agree. Changed with the last commit. 

> Regarding the startup message, you refer to this one?

Yes

> I think we can call s3_register("spatstat.linnet::as.linnet", "sfnetwork") inside suppressPackageStartupMessages() to avoid that, but not sure if that is good practice.

I added a call to `suppressPackageStartupMessages()`, feel free to remove that.

**Pull request merged**